ood-portal-generator/lib/ood_portal_generator/dex.rb

```diff
@@ -64,12 +62,27 @@ def self.installed?
       File.directory?(config_dir) && File.executable?('/usr/sbin/ondemand-dex')
     end
 
+    def self.default_auth
+      ['AuthType openid-connect', 'Require valid-user'].freeze
+    end
+
     def self.config_dir
       '/etc/ood/dex'
     end
 
     private
 
+    # determine if this config would enable dex configurations.
+    def enable?(config)
```

Comment:
I believe the public `enabled?` interface is used by the application and removing it may cause issues or require more refactoring in our tests.

That said, I will look into refactoring this a bit because `enabled?` and this `enable?` is a bit confusing.  And honestly, I think if we just use a different default (`nil` instead of `{}` for dex's config) we may not need/want this block at all.

Reply:
I fixed this as much as i could to actually set the member varialbe in `set_enable`.